In [3]:
import pandas as pd
import numpy as np
import seaborn as sns
np.random.seed(42)

#### Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [63]:
# Generating the data 
data = {'hours': [5,6,4,7,8,5,4,9,7,9], 'score': [88,76,69,81,90,71,62,93,82,95]}
df = pd.DataFrame(data = data)
df.head()

hours  score
0      5     88
1      6     76
2      4     69
3      7     81
4      8     90

In [66]:
df['hours'].corr(df['score'], method = 'pearson')

0.8644334560458191

#### Interpretation : The pearson relation of coefficient for studying hours of student and their exam score comes out to be r = 0.8644 which signifies high correlation which means that when students study for long hours their performance in exam tends to be better. When study hours increases, exam score increases too in same direction. This is clearly the example of positive linear relaionship between the hours and score variable.  

#### Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [77]:
# Data generation 
data = {'sleep_hours': [7,8,5,6,7,6], 'job_satisfaction': [7,9,5,7,7,5]}
df = pd.DataFrame(data = data)

In [71]:
df['sleep_hours'].corr(df['job_satisfaction'], method = 'spearman')

0.8740073734751262

#### Interpretation : The Spearman rank correlation coefficient for sleeping hours of individuals and their job satisfaction comes out to be r = 0.8740 which signifies high positive correlation which means that when individulas take enough sleep their performance and job satisfaction tends to be better.

#### Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [97]:
# Data generation :
np.random.seed(42)

hours = np.random.randint(5,10,50)
bmi = np.random.randint(18, 30, 50)
data = {'hours': hours, 'bmi': bmi}
df = pd.DataFrame(data = data)

In [98]:
# Pearson Correlation coefficient 

df['hours'].corr(df['bmi'], method = 'pearson')

0.18819692827827827

In [99]:
# spearman's rank correlation coefficient 

df['hours'].corr(df['bmi'], method = 'spearman')

0.20357968858475514

#### Interpretation: We can see from the above obseravations that spearman's and pearson correlation coefficient for exercise hours per week and bmi comes out to be almost same with spearmans correlation having slight high value, this means that our correlation is monotonic in nature. Apart from this both values signifies less to negligable correlation between the two factors into consideration. 

#### Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [104]:
# here we have two variables hours per day and level of physical activity. 
# We will take level of physical activity in range of 1-5. where value close to 5 signifies high level of activity and 
# if close to 1 siginifies very low level of activity.

hours = np.random.randint(1, 4, 50)
activity_level = np.random.randint(1, 5, 50)

data = {'hours': hours, 'activity_level': activity_level}
df = pd.DataFrame(data = data)
df.head()

hours  activity_level
0      2               3
1      1               2
2      2               1
3      3               2
4      2               2

In [106]:
df['hours'].corr(df['activity_level'], method = 'pearson')

-0.045662972460075106

#### Interpretation: The pearson correlation coefficient comes out to be -0.04566 which signifies low to negligable level of correlation between both factors. 

#### Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

In [4]:
data = {'age' : [25,42,37,19,31,28], 'drink' : ['Coke', 'Pepsi', 'Mountain dew', 'Coke', 'Pepsi', 'Coke']}

df = pd.DataFrame(data)

In [11]:
from sklearn.preprocessing import OneHotEncoder
encode = OneHotEncoder()
encoded = encode.fit_transform(df[['drink']])

In [21]:
encoded_df = pd.DataFrame(encoded.toarray(), columns = encode.get_feature_names_out())

In [23]:
df1 = pd.concat([df, encoded_df], axis = 1)

In [31]:
age_coke_corr = df1['age'].corr(df1['drink_Coke'])
age_dew_corr = df1['age'].corr(df1['drink_Mountain dew'])
age_pepsi_corr = df1['age'].corr(df1['drink_Pepsi'])

In [34]:
print("""The coefficient of correlation for age and coke drink comes out to be -0.837 which is sufficiently high degree of negative correlation.""") 
print("""The coefficient of correlation for age and MOuntain dew drink comes out to be 0.394 which is sufficiently less positively correlated.""") 
print("""The coefficient of correlation for age and Pepsi drink comes out to be 0.576 which is moderately positively correlated.""") 

The coefficient of correlation for age and coke drink comes out to be -0.837 which is sufficiently high degree of negative correlation.
The coefficient of correlation for age and MOuntain dew drink comes out to be 0.394 which is sufficiently less positively correlated.
The coefficient of correlation for age and Pepsi drink comes out to be 0.576 which is moderately positively correlated.


#### Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [156]:
num_calls = np.random.randint(1, 30, 30)
num_sales = np.random.randint(5, 50, 30)

In [157]:
data = {'num_calls': num_calls, 'num_sales': num_sales}
df = pd.DataFrame(data = data)
df.head()

num_calls  num_sales
0         23         39
1         10         41
2         12          9
3         14          5
4         14         44

In [158]:
df['num_calls'].corr(df['num_sales'])

0.04792424546518799

#### Interpretation: The pearson correlation coefficient comes out to be 0.04792 which signifies low to negligable positive correlation between factors.